js/apps/admin-ui/src/authentication/components/ExecutionConfigModal.tsx

```diff
@@ -54,22 +54,52 @@ export const ExecutionConfigModal = ({
     formState: { errors },
   } = form;
 
+  // default config all executions should have
+  const defaultConfigProperties = [
+    {
+      helpText: t("authenticatorRefConfig.help"),
+      label: t("authenticatorRefConfig.label"),
+      name: "default.reference",
+      readOnly: false,
+      secret: false,
+      type: "String",
+    },
+  ];
+
   const setupForm = (config?: AuthenticatorConfigRepresentation) => {
     convertToFormValues(config || {}, setValue);
   };
 
   useFetch(
     async () => {
-      const configDescription =
-        await adminClient.authenticationManagement.getConfigDescription({
-          providerId: execution.providerId!,
-        });
       let config: AuthenticatorConfigRepresentation | undefined;
+      let configDescription: AuthenticatorConfigInfoRepresentation;
+
+      // fetch execution config if its configurable
+      if (execution.configurable) {
```

Comment:
If you need to assign a value with two branches, you can use a ternary operator to differentiate them and do a direct assignment.

```ts
const foo = condition ? a : b;
```

Reply:
Thats much cleaner, i will update that when i update the PR based on @mposolda 's feedback below